xformers/helpers/hierarchical_configs.py

```diff
@@ -61,43 +62,49 @@ def get_hierarchical_configuration(
             "add_class_token": False,
         },
         "patch_embedding_config": {
-            "in_channels": 3,
+            "in_channels": in_channels,
             "kernel_size": 0,
             "stride": 0,
             "padding": 0,
         },
     }
 
     xformers_config = []
-    in_channels = 3
+    in_channels = in_channels
 
-    for layer_basic_config in layer_basic_configs:
```

Comment:
minor renaming to try to make it easier to read. The main idea is that we have a "base" config for the layers, and that this util fills in the gaps so that we can completely specify the model